contributors/design-proposals/taint-node-by-condition.md

```diff
@@ -0,0 +1,41 @@
+# Taints Node according to NodeConditions
+
+@k82cn, @gmarek, @jamiehannaford,  Jul 15, 2017
+
+## Relevant issues:
+
+https://github.com/kubernetes/kubernetes/issues/42001
+https://github.com/kubernetes/kubernetes/issues/45717
+
+## Motivation
+In kubernetes 1.8 and before, there are six Node Conditions, each with three possible values: True, False or Unknown. The k8s components check those conditions without Pods’ attributes, e.g. the scheduler will filter out all nodes whose NetworkUnavailable condition is True, meaning that pods on the host network can not be scheduled to those nodes, even though a user might want that.  The motivation of this proposal is to taint Nodes based on certain conditions, so that other components can leverage Tolerations for more advanced scheduling.
```

Comment:
Kubernetes components modify and check those node conditions without any consideration to pods and their specs. For example, the scheduler currently filters out all nodes whose NetworkUnavailable condition is true ...